lib/sshkit/ssh/dry_run/ssh_connection.ex

```diff
@@ -0,0 +1,34 @@
+defmodule SSHKit.SSH.DryRun.SSHConnection do
+  @moduledoc false
+
+  require Logger
+
+  @ssh_channel_id :dry_run_channel
+  def session_channel(_ref, _ini_window_size, _max_packet_size, _timeout) do
+    {:ok, @ssh_channel_id}
+  end
+
+  def close(_ref, _id) do
```

Comment:
Only a question of style, but I'd say, for these "no_op" functions, we could use the shorter one-line syntax to emphasize them not doing much:
```elixir
def close(_ref, _id), do: :ok
```